samples/bluetooth/central_uart/src/main.c

```diff
@@ -478,5 +492,10 @@ void main(void)
 			printk("Failed to send data over BLE connection"
 			       "(err %d)\n", err);
 		}
+
+		err = k_sem_take(&nus_write_sem, NUS_WRITE_TIMEOUT);
```

Comment:
Are you sure that 150 timeout is enough? Did you consider using semaphore without any timeout restriction if ```ble_data_sent``` callback is reliable?

Reply:
I choose this time experimentally for this data amount. I also considered taking a semaphore with `K_FOREVER` but I think if the Bluetooth hang, application can received UART messages and try to send it while heap memory is not full. Exceeding this time causes only displaying a message with timeout information.